In [1]:
import os
import nltk
import string

import tensorflow as tf
from tensorflow.keras import layers
from tensorflow.keras import models

from gensim.models import Word2Vec

from util.preprocessing import *

## Reviews dataset v2

In [2]:
PATH = "./data/txt_sentoken/"

In [3]:
docs = []

In [4]:
for direc in os.listdir(PATH):
    for file in os.listdir(PATH + direc):
        try:
            with open(PATH + direc + "/" + file, "r") as f:
                docs.append(
                    f.read()
                )
        except:
            continue

In [5]:
len(docs)

2000

In [6]:
word_set, phrases = process_documents(docs)

100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2000/2000 [00:28<00:00, 69.55it/s]


In [7]:
len(word_set)

39212

## Reviews dataset v1

In [9]:
PATH = "./data/rt-polaritydata/"

In [14]:
docs = []

In [15]:
for file in os.listdir(PATH):
    try:
        with open(PATH + file, "r") as f:
            docs+=(
                f.readlines()
            )
    except:
        continue

In [17]:
len(docs)

10662

In [18]:
word_set, phrases = process_documents(docs)

100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10662/10662 [00:06<00:00, 1611.66it/s]


In [19]:
len(word_set)

18190

We can see that the most recent dataset as $39212$ distinct words while the 1st version has only $18190$.

In the paper (where the 1st version was used) they stated to have $18765$ words in the dataset. It is almost the same has what we obtained. 
The algorithm used to filter or to tokenize the words must be slightly different.

# Modelling

In [5]:
dataset, validation = tf.keras.utils.text_dataset_from_directory(
    PATH, 
    validation_split=0.2, 
    subset="both",
    seed=13
)

Found 2002 files belonging to 2 classes.
Using 1602 files for training.
Using 400 files for validation.
